hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/NamespacesInstanceResource.java

```diff
@@ -135,35 +135,9 @@ public Response get(final @Context ServletContext context,
   @Consumes({MIMETYPE_XML, MIMETYPE_JSON, MIMETYPE_PROTOBUF,
     MIMETYPE_PROTOBUF_IETF})
   public Response put(final NamespacesInstanceModel model, final @Context UriInfo uriInfo) {
-    if (LOG.isTraceEnabled()) {
-      LOG.trace("PUT " + uriInfo.getAbsolutePath());
-    }
-    servlet.getMetrics().incrementRequests(1);
     return processUpdate(model, true, uriInfo);
   }
 
-  /**
-   * Build a response for PUT alter namespace with no properties specified.
-   * @param message value not used.
-   * @param headers value not used.
-   * @return response code.
-   */
-  @PUT
-  public Response putNoBody(final byte[] message,
```

Comment:
I don't know how this works. When you remove this, the put gets called?

Reply:
For some reason, on branch-2 based versions, these putNoBody and postNoBody methods only get properly called if client sets content-type request header to "application/octet-stream". We don't explicit define any requirement for "binary" request content-type, on the REST API doc in the ref guide, so it seems clients had been working well without setting content-type, in hbase 1. I suspect the behaviour change might be due jersey version changes between branch-1 and branch-2. 
This fix simply removed the additional method, so request is guaranteed to always go to same place and then payload validations are applied there.